packages/blaze/template.js

```diff
@@ -56,6 +62,52 @@ Blaze.isTemplate = function (t) {
   return (t instanceof Blaze.Template);
 };
 
+/**
+ * @name  onCreated
+ * @instance
+ * @memberOf Template
+ * @summary Add a callback when an instance of a template is created.
+ * @param {Function} callback A function to be added as a callback.
+ * @locus Client
+ */
+Template.prototype.onCreated = function (cb) {
+  this._callbacks.created.push(cb);
+};
+
+/**
+ * @name  onRendered
+ * @instance
+ * @memberOf Template
+ * @summary Add a callback when an instance of a template is rendered.
+ * @param {Function} callback A function to be added as a callback.
+ * @locus Client
+ */
+Template.prototype.onRendered = function (cb) {
+  this._callbacks.rendered.push(cb);
+};
+
+/**
+ * @name  onDestroyed
+ * @instance
+ * @memberOf Template
+ * @summary Add a callback when an instance of a template is destroyed.
+ * @param {Function} callback A function to be added as a callback.
+ * @locus Client
+ */
+Template.prototype.onDestroyed = function (cb) {
+  this._callbacks.destroyed.push(cb);
+};
+
+Template.prototype._fireCallbacks = function (template, which) {
+  var self = this;
+  var callbacks = self[which] ? [self[which]] : [];
```

Comment:
Could use a comment. So we fire the .rendered callback _or_ .onRendered? But not both? Why not?